browser/extensions/api/ipfs_api.cc

```diff
@@ -77,8 +77,7 @@ ExtensionFunction::ResponseAction IpfsRemoveIpnsKeyFunction::Run() {
   EXTENSION_FUNCTION_VALIDATE(params.get());
   key_manager->RemoveKey(
       params->name, base::BindOnce(&IpfsRemoveIpnsKeyFunction::OnKeyRemoved,
-                                   base::Unretained(this), key_manager));
-  AddRef();
+                                   base::RetainedRef(this), key_manager));
   return RespondLater();
```

Comment:
pls double-check the control flow in all async calls. Sometimes `RemoveKey` invokes the callback synchronously (with immediate `Respond()`), so you probably want something like 
`  return did_respond() ? AlreadyResponded() : RespondLater();`